src/main/java/org/motechproject/newebodac/service/VisitTypeService.java

```diff
@@ -30,13 +31,24 @@ public VisitTypeDto create(VisitTypeDto visitTypeDto) {
     return MAPPER.toDto(visitTypeRepository.save(MAPPER.fromDto(visitTypeDto)));
   }
 
+  public VisitTypeDto update(UUID id, VisitTypeDto visitTypeDto) {
+    VisitType visitType = visitTypeRepository.findById(id)
+        .orElseThrow(() -> new EntityNotFoundException("Visit type with id: {0} not found",
+            id.toString()));
+    MAPPER.update(visitTypeDto, visitType);
+    return MAPPER.toDto(visitTypeRepository.save(visitType));
+  }
+
   /**
    * Deletes visit type with given id.
    * @param id ID of visit type to delete.
    */
   public void delete(UUID id) {
     VisitType visitType = visitTypeRepository.findById(id).orElseThrow(() ->
         new EntityNotFoundException("Visit Type with id: {0} not found", id.toString()));
+    if (!visitType.getMessages().isEmpty()) {
```

Comment:
I'm not sure about this one, I think it should be possible to remove visit type with all the messages (you may add cascade delete and orphan removal there), but it should not be possible to remove visit type if any visit is using it